ath79: add support for Senao Engenius ENS1750

abraha2d opened the pull request:
FCC ID: A8J-EWS660AP

Engenius ENS1750 is an outdoor wireless access point with 2 gigabit ethernet ports, dual-band wireless,
internal antenna plates, and 802.3at PoE+

Engenius EWS660AP, ENS1750, and ENS1200 are "electrically identical, different model names are for marketing purpose" according to docs provided by Engenius to the FCC.

**Specification:**

  - QCA9558 SOC		2.4 GHz, 3x3
  - QCA9880 WLAN	mini PCIe card, 5 GHz, 3x3, 26dBm
  - AR8035-A PHY	RGMII GbE with PoE+ IN
  - AR8033 PHY		SGMII GbE with PoE+ OUT
  - 40 MHz clock
  - 16 MB FLASH		MX25L12845EMI-10G
  - 2x 64 MB RAM
  - UART at J1		populated, RX grounded
  - 6 internal antenna plates (5 dbi, omni-directional)
  - 5 LEDs, 1 button (power, eth0, eth1, 2G, 5G) (reset)

**MAC addresses:**

  Base MAC addressed labeled as "MAC"
  Only one Vendor MAC address in flash

  eth0 *:d4 MAC art 0x0
  eth1 *:d5 --- art 0x0 +1
  phy1 *:d6 --- art 0x0 +2
  phy0 *:d7 --- art 0x0 +3

**Serial Access:**

  the RX line on the board for UART is shorted to ground by resistor R176
  therefore it must be removed to use the console
  but it is not necessary to remove to view boot log

  optionally, R175 can be replaced with a solder bridge short

  the resistors R175 and R176 are next to the UART RX pin

**Installation:**

  2 ways to flash factory.bin from OEM:

  Method 1: Firmware upgrade page:

    OEM webpage at 192.168.1.1 username and password "admin" Navigate to "Firmware Upgrade" page from left pane Click Browse and select the factory.bin image Upload and verify checksum Click Continue to confirm and wait 3 minutes

  Method 2: Serial to load Failsafe webpage:

    After connecting to serial console and rebooting... Interrupt uboot with any key pressed rapidly execute `run failsafe_boot` OR `bootm 0x9fd70000` wait a minute connect to ethernet and navigate to "192.168.1.1/index.htm" Select the factory.bin image and upload wait about 3 minutes

**Return to OEM:**

  If you have a serial cable, see Serial Failsafe instructions
  otherwise, uboot-env can be used to make uboot load the failsafe image

  ssh into openwrt and run
  `fw_setenv rootfs_checksum 0`
  reboot, wait 3 minutes
  connect to ethernet and navigate to 192.168.1.1/index.htm
  select OEM firmware image from Engenius and click upgrade

**TFTP recovery:**

  Requires serial console, reset button does nothing

  rename initramfs.bin to '0101A8C0.img'
  make available on TFTP server at 192.168.1.101
  power board, interrupt boot
  execute tftpboot and bootm 0x81000000

**Format of OEM firmware image:**

  The OEM software of ENS1750 is a heavily modified version
  of Openwrt Kamikaze. One of the many modifications
  is to the sysupgrade program. Image verification is performed
  simply by the successful ungzip and untar of the supplied file
  and name check and header verification of the resulting contents.
  To form a factory.bin that is accepted by OEM Openwrt build,
  the kernel and rootfs must have specific names...

    openwrt-ar71xx-generic-ens1750-uImage-lzma.bin openwrt-ar71xx-generic-ens1750-root.squashfs

  and begin with the respective headers (uImage, squashfs).
  Then the files must be tarballed and gzipped.
  The resulting binary is actually a tar.gz file in disguise.
  This can be verified by using binwalk on the OEM firmware images,
  ungzipping then untaring.

  Newer EnGenius software requires more checks but their script
  includes a way to skip them, otherwise the tar must include
  a text file with the version and md5sums in a deprecated format.

  The OEM upgrade script is at /etc/fwupgrade.sh.

  OKLI kernel loader is required because the OEM software
  expects the kernel to be no greater than 1536k
  and the factory.bin upgrade procedure would otherwise
  overwrite part of the kernel when writing rootfs.

Note on PLL-data cells:

  The default PLL register values will not work
  because of the external AR8035 switch between
  the SOC and the ethernet port.

  For QCA955x series, the PLL registers for eth0 and eth1
  can be see in the DTSI as 0x28 and 0x48 respectively.
  Therefore the PLL registers can be read from uboot
  for each link speed after attempting tftpboot
  or another network action using that link speed
  with `md 0x18050028 1` and `md 0x18050048 1`.

  The clock delay required for RGMII can be applied
  at the PHY side, using the at803x driver `phy-mode`.
  Therefore the PLL registers for GMAC0
  do not need the bits for delay on the MAC side.
  This is possible due to fixes in at803x driver
  since Linux 5.1 and 5.3

Tested-by: Kevin Abraham <kevin@westhousefarm.com>

abraha2d committed:
ath79: add support for Senao Engenius ENS1750

FCC ID: A8J-EWS660AP

Engenius ENS1750 is an outdoor wireless access point with
2 gigabit ethernet ports, dual-band wireless,
internal antenna plates, and 802.3at PoE+

Engenius EWS660AP, ENS1750, and ENS1200 are "electrically identical,
different model names are for marketing purpose" according to docs
provided by Engenius to the FCC.

**Specification:**

  - QCA9558 SOC		2.4 GHz, 3x3
  - QCA9880 WLAN	mini PCIe card, 5 GHz, 3x3, 26dBm
  - AR8035-A PHY	RGMII GbE with PoE+ IN
  - AR8033 PHY		SGMII GbE with PoE+ OUT
  - 40 MHz clock
  - 16 MB FLASH		MX25L12845EMI-10G
  - 2x 64 MB RAM
  - UART at J1		populated, RX grounded
  - 6 internal antenna plates (5 dbi, omni-directional)
  - 5 LEDs, 1 button (power, eth0, eth1, 2G, 5G) (reset)

**MAC addresses:**

  Base MAC addressed labeled as "MAC"
  Only one Vendor MAC address in flash

  eth0 *:d4 MAC art 0x0
  eth1 *:d5 --- art 0x0 +1
  phy1 *:d6 --- art 0x0 +2
  phy0 *:d7 --- art 0x0 +3

**Serial Access:**

  the RX line on the board for UART is shorted to ground by resistor R176
  therefore it must be removed to use the console
  but it is not necessary to remove to view boot log

  optionally, R175 can be replaced with a solder bridge short

  the resistors R175 and R176 are next to the UART RX pin

**Installation:**

  2 ways to flash factory.bin from OEM:

  Method 1: Firmware upgrade page:

    OEM webpage at 192.168.1.1
    username and password "admin"
    Navigate to "Firmware Upgrade" page from left pane
    Click Browse and select the factory.bin image
    Upload and verify checksum
    Click Continue to confirm and wait 3 minutes

  Method 2: Serial to load Failsafe webpage:

    After connecting to serial console and rebooting...
    Interrupt uboot with any key pressed rapidly
    execute `run failsafe_boot` OR `bootm 0x9fd70000`
    wait a minute
    connect to ethernet and navigate to
    "192.168.1.1/index.htm"
    Select the factory.bin image and upload
    wait about 3 minutes

**Return to OEM:**

  If you have a serial cable, see Serial Failsafe instructions
  otherwise, uboot-env can be used to make uboot load the failsafe image

  ssh into openwrt and run
  `fw_setenv rootfs_checksum 0`
  reboot, wait 3 minutes
  connect to ethernet and navigate to 192.168.1.1/index.htm
  select OEM firmware image from Engenius and click upgrade

**TFTP recovery:**

  Requires serial console, reset button does nothing

  rename initramfs.bin to '0101A8C0.img'
  make available on TFTP server at 192.168.1.101
  power board, interrupt boot
  execute tftpboot and bootm 0x81000000

**Format of OEM firmware image:**

  The OEM software of ENS1750 is a heavily modified version
  of Openwrt Kamikaze. One of the many modifications
  is to the sysupgrade program. Image verification is performed
  simply by the successful ungzip and untar of the supplied file
  and name check and header verification of the resulting contents.
  To form a factory.bin that is accepted by OEM Openwrt build,
  the kernel and rootfs must have specific names...

    openwrt-ar71xx-generic-ens1750-uImage-lzma.bin
    openwrt-ar71xx-generic-ens1750-root.squashfs

  and begin with the respective headers (uImage, squashfs).
  Then the files must be tarballed and gzipped.
  The resulting binary is actually a tar.gz file in disguise.
  This can be verified by using binwalk on the OEM firmware images,
  ungzipping then untaring.

  Newer EnGenius software requires more checks but their script
  includes a way to skip them, otherwise the tar must include
  a text file with the version and md5sums in a deprecated format.

  The OEM upgrade script is at /etc/fwupgrade.sh.

  OKLI kernel loader is required because the OEM software
  expects the kernel to be no greater than 1536k
  and the factory.bin upgrade procedure would otherwise
  overwrite part of the kernel when writing rootfs.

Note on PLL-data cells:

  The default PLL register values will not work
  because of the external AR8035 switch between
  the SOC and the ethernet port.

  For QCA955x series, the PLL registers for eth0 and eth1
  can be see in the DTSI as 0x28 and 0x48 respectively.
  Therefore the PLL registers can be read from uboot
  for each link speed after attempting tftpboot
  or another network action using that link speed
  with `md 0x18050028 1` and `md 0x18050048 1`.

  The clock delay required for RGMII can be applied
  at the PHY side, using the at803x driver `phy-mode`.
  Therefore the PLL registers for GMAC0
  do not need the bits for delay on the MAC side.
  This is possible due to fixes in at803x driver
  since Linux 5.1 and 5.3

Tested-by: Kevin Abraham <kevin@westhousefarm.com>
Signed-off-by: Kevin Abraham <kevin@westhousefarm.com>

abraha2d commented:
Note: as mentioned above, the ENS1750 is identical to the EWS660AP in everything but name.
Therefore, this PR is also identical in everything but name to @mpratt14's PR adding support for the EWS660AP: https://github.com/openwrt/openwrt/pull/11792
 
Please let me know if there is a better way of adding support for the ENS1750 given its similarities to the EWS660AP.
Firmware built for the EWS660AP will not boot on the ENS1750 due to the difference in names.

abraha2d commented:
Kernel boot log:
```
[    0.000000] Linux version 5.15.150 (builder@buildhost) (mips-openwrt-linux-musl-gcc (OpenWrt GCC 12.3.0 r23809-234f1a2efa) 12.3.0, GNU ld (GNU Binutils) 2.40.0) #0 Fri Mar 22 22:09:42 2024
[    0.000000] printk: bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019750 (MIPS 74Kc)
[    0.000000] MIPS: machine is EnGenius ENS1750
[    0.000000] SoC: Qualcomm Atheros QCA9558 ver 1 rev 0
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x000000000fffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x000000000fffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x000000000fffffff]
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 64960
[    0.000000] Kernel command line: console=ttyS0,115200n8 rootfstype=squashfs,jffs2
[    0.000000] Dentry cache hash table entries: 32768 (order: 5, 131072 bytes, linear)
[    0.000000] Inode-cache hash table entries: 16384 (order: 4, 65536 bytes, linear)
[    0.000000] Writing ErrCtl register=00000000
[    0.000000] Readback ErrCtl register=00000000
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 250704K/262144K available (6057K kernel code, 598K rwdata, 780K rodata, 1196K init, 216K bss, 11440K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 51
[    0.000000] CPU clock: 720.000 MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 5309056796 ns
[    0.000001] sched_clock: 32 bits at 360MHz, resolution 2ns, wraps every 5965232126ns
[    0.008355] Calibrating delay loop... 358.80 BogoMIPS (lpj=1794048)
[    0.075032] pid_max: default: 32768 minimum: 301
[    0.080827] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.088620] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.102994] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.113499] futex hash table entries: 256 (order: -1, 3072 bytes, linear)
[    0.120959] pinctrl core: initialized pinctrl subsystem
[    0.128193] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.135046] thermal_sys: Registered thermal governor 'step_wise'
[    0.146873] clocksource: Switched to clocksource MIPS
[    0.160025] NET: Registered PF_INET protocol family
[    0.165440] IP idents hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.174147] tcp_listen_portaddr_hash hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.183195] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.191460] TCP established hash table entries: 2048 (order: 1, 8192 bytes, linear)
[    0.199645] TCP bind hash table entries: 2048 (order: 1, 8192 bytes, linear)
[    0.207179] TCP: Hash tables configured (established 2048 bind 2048)
[    0.214077] UDP hash table entries: 256 (order: 0, 4096 bytes, linear)
[    0.221093] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes, linear)
[    0.228917] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.234978] PCI: CLS 0 bytes, default 32
[    0.242192] workingset: timestamp_bits=14 max_order=16 bucket_order=2
[    0.254082] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.260350] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.272031] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 252)
[    0.283525] pinctrl-single 1804002c.pinmux: 544 pins, size 68
[    0.290819] Serial: 8250/16550 driver, 16 ports, IRQ sharing enabled
[    0.300006] printk: console [ttyS0] disabled
[    0.304618] 18020000.uart: ttyS0 at MMIO 0x18020000 (irq = 9, base_baud = 2500000) is a 16550A
[    0.313852] printk: console [ttyS0] enabled
[    0.322754] printk: bootconsole [early0] disabled
[    0.349422] spi-nor spi0.0: mx25l12805d (16384 Kbytes)
[    0.354706] 8 fixed-partitions partitions found on MTD device spi0.0
[    0.361233] OF: Bad cell count for /ahb/spi@1f000000/flash@0/partitions
[    0.368000] OF: Bad cell count for /ahb/spi@1f000000/flash@0/partitions
[    0.375411] Creating 8 MTD partitions on "spi0.0":
[    0.380343] 0x000000000000-0x000000040000 : "u-boot"
[    0.390356] 0x000000040000-0x000000050000 : "u-boot-env"
[    0.396625] 0x000000050000-0x0000000a0000 : "custom"
[    0.404094] 0x0000000a0000-0x0000000b0000 : "loader"
[    0.410112] 0x0000000b0000-0x000000220000 : "fwconcat1"
[    0.417809] 0x000000220000-0x000000d70000 : "fwconcat0"
[    0.424071] 0x000000d70000-0x000000ff0000 : "failsafe"
[    0.431652] 0x000000ff0000-0x000001000000 : "art"
[    0.440814] Concatenating MTD devices:
[    0.444632] (0): "fwconcat0"
[    0.447591] (1): "fwconcat1"
[    0.450508] into device "virtual_flash"
[    0.454427] 1 fixed-partitions partitions found on MTD device virtual_flash
[    0.461747] Creating 1 MTD partitions on "virtual_flash":
[    0.467297] 0x000000000000-0x000000cc0000 : "firmware"
[    0.480641] 2 uimage-fw partitions found on MTD device firmware
[    0.486662] Creating 2 MTD partitions on "firmware":
[    0.491746] 0x000000000000-0x000000240000 : "kernel"
[    0.498433] 0x000000240000-0x000000cc0000 : "rootfs"
[    0.504292] mtd: setting mtd10 (rootfs) as root device
[    0.510345] 1 squashfs-split partitions found on MTD device rootfs
[    0.516641] 0x000000610000-0x000000cc0000 : "rootfs_data"
[    0.987468] ag71xx 19000000.eth: connected to PHY at mdio.0:01 [uid=004dd072, driver=Qualcomm Atheros AR8035]
[    0.998127] eth0: Atheros AG71xx at 0xb9000000, irq 4, mode: rgmii-id
[    1.427628] ag71xx 1a000000.eth: connected to PHY at mdio.0:02 [uid=004dd074, driver=Qualcomm Atheros AR8031/AR8033]
[    1.438893] eth1: Atheros AG71xx at 0xba000000, irq 5, mode: sgmii
[    1.445538] i2c_dev: i2c /dev entries driver
[    1.452068] NET: Registered PF_INET6 protocol family
[    1.464645] Segment Routing with IPv6
[    1.468500] In-situ OAM (IOAM) with IPv6
[    1.472598] NET: Registered PF_PACKET protocol family
[    1.477858] bridge: filtering via arp/ip/ip6tables is no longer available by default. Update your scripts to load br_netfilter if you need this.
[    1.491301] 8021q: 802.1Q VLAN Support v1.8
[    1.496717] PCI host bridge to bus 0000:00
[    1.500931] pci_bus 0000:00: root bus resource [mem 0x12000000-0x13ffffff]
[    1.507923] pci_bus 0000:00: root bus resource [io  0x0001]
[    1.513577] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    1.521671] pci 0000:00:00.0: [168c:003c] type 00 class 0x028000
[    1.527818] pci 0000:00:00.0: reg 0x10: [mem 0x00000000-0x001fffff 64bit]
[    1.534750] pci 0000:00:00.0: reg 0x30: [mem 0x00000000-0x0000ffff pref]
[    1.541636] pci 0000:00:00.0: supports D1
[    1.545704] pci 0000:00:00.0: PME# supported from D0 D1 D3hot
[    1.552757] pci_bus 0000:00: busn_res: [bus 00-ff] end is updated to 00
[    1.559540] pci 0000:00:00.0: BAR 0: assigned [mem 0x12000000-0x121fffff 64bit]
[    1.566993] pci 0000:00:00.0: BAR 6: assigned [mem 0x12200000-0x1220ffff pref]
[    1.583651] VFS: Mounted root (squashfs filesystem) readonly on device 31:10.
[    1.595043] Freeing unused kernel image (initmem) memory: 1196K
[    1.601086] This architecture does not have kernel memory protection.
[    1.607632] Run /sbin/init as init process
[    1.611783]   with arguments:
[    1.611787]     /sbin/init
[    1.611792]   with environment:
[    1.611796]     HOME=/
[    1.611801]     TERM=linux
[    2.221468] init: Console is alive
[    2.225366] init: - watchdog -
[    3.473836] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    3.543482] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    3.561308] init: - preinit -
[    5.068570] random: jshn: uninitialized urandom read (4 bytes read)
[    5.479758] random: jshn: uninitialized urandom read (4 bytes read)
[    5.516962] random: jshn: uninitialized urandom read (4 bytes read)
[    8.005749] jffs2: notice: (417) jffs2_build_xattr_subsystem: complete building xattr subsystem, 16 of xdatum (0 unchecked, 5 orphan) and 18 of xref (2 dead, 3 orphan) found.
[    8.023218] mount_root: switching to jffs2 overlay
[    8.032872] overlayfs: upper fs does not support tmpfile.
[    8.047640] urandom-seed: Seeding with /etc/urandom.seed
[    8.188834] procd: - early -
[    8.192039] procd: - watchdog -
[    8.926266] procd: - watchdog -
[    8.930835] procd: - ubus -
[    9.047761] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.056433] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.069978] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.080963] procd: - init -
[   10.116019] random: jshn: uninitialized urandom read (4 bytes read)
[   10.188615] random: ubusd: uninitialized urandom read (4 bytes read)
[   10.196814] random: ubus: uninitialized urandom read (4 bytes read)
[   10.250835] kmodloader: loading kernel modules from /etc/modules.d/*
[   10.829008] Loading modules backported from Linux version v6.1.24-0-g0102425ac76b
[   10.836613] Backport generated by backports.git v5.15.92-1-44-gd6ea70fafd36
[   11.479493] PPP generic driver version 2.4.2
[   11.487089] NET: Registered PF_PPPOX protocol family
[   11.549773] ath10k 5.15 driver, optimized for CT firmware, probing pci device: 0x3c.
[   11.567858] ath10k_pci 0000:00:00.0: enabling device (0000 -> 0002)
[   11.574401] ath10k_pci 0000:00:00.0: pci irq legacy oper_irq_mode 1 irq_mode 0 reset_mode 0
[   12.537344] urngd: v1.0.2 started.
[   14.397205] random: crng init done
[   14.400666] random: 25 urandom warning(s) missed due to ratelimiting
[   17.074317] ath10k_pci 0000:00:00.0: qca988x hw2.0 target 0x4100016c chip_id 0x043202ff sub 0000:0000
[   17.083733] ath10k_pci 0000:00:00.0: kconfig debug 0 debugfs 1 tracing 0 dfs 1 testmode 0
[   17.095654] ath10k_pci 0000:00:00.0: firmware ver 10.1-ct-8x-__fW-022-ecad3248 api 2 features wmi-10.x,has-wmi-mgmt-tx,mfp,txstatus-noack,wmi-10.x-CT,ratemask-CT,txrate-CT,get-temp-CT,tx-rc-CT,cust-stats-CT,retry-gt2-CT,txrate2-CT,beacon-cb-CT,wmi-block-ack-CT crc32 3e4cf97f
[   17.207947] ath10k_pci 0000:00:00.0: Loading BDF type 0
[   18.118276] ath10k_pci 0000:00:00.0: board_file api 1 bmi_id N/A crc32 bebc7c08
[   19.133728] ath10k_pci 0000:00:00.0: 10.1 wmi init: vdevs: 16  peers: 127  tid: 256
[   19.150386] ath10k_pci 0000:00:00.0: wmi print 'P 128 V 8 T 410'
[   19.156683] ath10k_pci 0000:00:00.0: wmi print 'msdu-desc: 1424  sw-crypt: 0 ct-sta: 0'
[   19.165066] ath10k_pci 0000:00:00.0: wmi print 'alloc rem: 24984 iram: 38672'
[   19.230803] ath10k_pci 0000:00:00.0: htt-ver 2.1 wmi-op 2 htt-op 2 cal nvmem max-sta 128 raw 0 hwcrypto 1
[   19.240893] ath10k_pci 0000:00:00.0: NOTE:  Firmware DBGLOG output disabled in debug_mask: 0x10000000
[   19.410528] ath: EEPROM regdomain: 0x0
[   19.410547] ath: EEPROM indicates default country code should be used
[   19.410553] ath: doing EEPROM country->regdmn map search
[   19.410566] ath: country maps to regdmn code: 0x3a
[   19.410573] ath: Country alpha2 being used: US
[   19.410579] ath: Regpair used: 0x3a
[   19.486905] ath: EEPROM regdomain: 0x0
[   19.486927] ath: EEPROM indicates default country code should be used
[   19.486933] ath: doing EEPROM country->regdmn map search
[   19.486946] ath: country maps to regdmn code: 0x3a
[   19.486953] ath: Country alpha2 being used: US
[   19.486959] ath: Regpair used: 0x3a
[   19.504107] ieee80211 phy1: Selected rate control algorithm 'minstrel_ht'
[   19.505918] ieee80211 phy1: Atheros AR9550 Rev:0 mem=0xb8100000, irq=12
[   19.587345] kmodloader: done loading kernel modules from /etc/modules.d/*
[   36.949599] br-lan: port 1(eth0) entered blocking state
[   36.954917] br-lan: port 1(eth0) entered disabled state
[   36.960531] device eth0 entered promiscuous mode
[   37.009108] br-lan: port 2(eth1) entered blocking state
[   37.014431] br-lan: port 2(eth1) entered disabled state
[   37.020054] device eth1 entered promiscuous mode
[   40.098513] eth0: link up (1000Mbps/Full duplex)
[   40.103244] br-lan: port 1(eth0) entered blocking state
[   40.108581] br-lan: port 1(eth0) entered forwarding state
[   40.177034] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
[   41.577075] br-lan: port 3(phy1-ap0) entered blocking state
[   41.582749] br-lan: port 3(phy1-ap0) entered disabled state
[   41.588701] device phy1-ap0 entered promiscuous mode
[   41.763618] IPv6: ADDRCONF(NETDEV_CHANGE): phy1-ap0: link becomes ready
[   41.770529] br-lan: port 3(phy1-ap0) entered blocking state
[   41.776199] br-lan: port 3(phy1-ap0) entered forwarding state
[   43.062722] ath10k_pci 0000:00:00.0: 10.1 wmi init: vdevs: 16  peers: 127  tid: 256
[   43.079486] ath10k_pci 0000:00:00.0: wmi print 'P 128 V 8 T 410'
[   43.085794] ath10k_pci 0000:00:00.0: wmi print 'msdu-desc: 1424  sw-crypt: 0 ct-sta: 0'
[   43.094273] ath10k_pci 0000:00:00.0: wmi print 'alloc rem: 24984 iram: 38672'
[   43.161546] ath10k_pci 0000:00:00.0: pdev param 0 not supported by firmware
[   43.175887] ath10k_pci 0000:00:00.0: rts threshold -1
[   43.192501] br-lan: port 4(phy0-ap0) entered blocking state
[   43.198216] br-lan: port 4(phy0-ap0) entered disabled state
[   43.204164] device phy0-ap0 entered promiscuous mode
[   46.432712] IPv6: ADDRCONF(NETDEV_CHANGE): phy0-ap0: link becomes ready
[   46.439703] br-lan: port 4(phy0-ap0) entered blocking state
[   46.445388] br-lan: port 4(phy0-ap0) entered forwarding state
```

DragonBluep commented on package/boot/uboot-envtools/files/ath79:
```diff
@@ -41,6 +41,7 @@ engenius,ecb1750|\
 engenius,ecb350-v1|\
 engenius,ecb600|\
 engenius,enh202-v1|\
+engenius,ens1750|\
```
You can add a `compatible = "u-boot,env";` property to the device tree u-boot-env partition. This file is not necessary now. ref: https://github.com/openwrt/openwrt/commit/d7d94a8d91bc1dc9dcedbe004021509871acac24

abraha2d replied:
Done (for all devices using qca955x_senao_loader.dtsi). Left as a separate commit, let me know if it should be it's own PR due to the wide-reaching change.

DragonBluep commented on target/linux/ath79/dts/qca9558_engenius_ens1750.dts:
```diff
@@ -0,0 +1,133 @@
+		compatible = "gpio-leds";
+
+		wifi2g {
+			label = "green:wifi2g";
```
```suggestion
			color = <LED_COLOR_ID_GREEN>;
			function = LED_FUNCTION_WLAN_2GHZ;
```

DragonBluep commented on target/linux/ath79/dts/qca9558_engenius_ens1750.dts:
```diff
@@ -0,0 +1,133 @@
+		};
+
+		led_wifi5g: wifi5g {
+			label = "green:wifi5g";
```
```suggestion
			color = <LED_COLOR_ID_GREEN>;
			function = LED_FUNCTION_WLAN_5GHZ;
```
And we need `#include <dt-bindings/leds/common.h>` header.

abraha2d committed:
uboot-envtools: ath79: remove env config for Senao Loader devices

uboot-envtools can automatically parse the 'u-boot,env' compatible string from the dts.

Signed-off-by: Kevin Abraham <kevin@westhousefarm.com>

abraha2d commented:
Let me know if I should also add ENS1200... according to FCC it's also electrically identical, but I don't have a device to test with.

mpratt14 commented:
we ought to have a device to test it on just in case... I might have one but I don't remember, it's been a while since I went through everything I bought 

by the way, since your branch is up to date with the main branch, a lot of the things that were mentioned that needed changes are probably already applied to the dts that you are basing this new one on, so to make sure you can do a quick:

`diff target/linux/ath79/dts/qca9558_engenius_ews660ap.dts target/linux/ath79/dts/qca9558_engenius_ens1750.dts`

mpratt14 commented:
wait nevermind I see you are turning it into a dtsi...

abraha2d commented:
Yup, and for reference, this is the diff between the ews660ap dts on `main` and the dtsi on this branch:
```diff
diff --git a/target/linux/ath79/dts/qca9558_engenius_ews660ap.dts b/target/linux/ath79/dts/qca9558_engenius_ews660ap.dtsi
index 9fa1927c1d..4badef9e5b 100644
--- a/target/linux/ath79/dts/qca9558_engenius_ews660ap.dts
+++ b/target/linux/ath79/dts/qca9558_engenius_ews660ap.dtsi
@@ -4,11 +4,9 @@
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
 
 / {
-       compatible = "engenius,ews660ap", "qca,qca9558";
-       model = "EnGenius EWS660AP";
-
        aliases {
                label-mac-device = &eth0;
                led-boot = &led_wifi5g;
@@ -31,13 +29,15 @@
                compatible = "gpio-leds";
 
                wifi2g {
-                       label = "green:wifi2g";
+                       color = <LED_COLOR_ID_GREEN>;
+                       function = LED_FUNCTION_WLAN_2GHZ;
                        gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
                        linux,default-trigger = "phy1tpt";
                };
 
                led_wifi5g: wifi5g {
-                       label = "green:wifi5g";
+                       color = <LED_COLOR_ID_GREEN>;
+                       function = LED_FUNCTION_WLAN_5GHZ;
                        gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
                        linux,default-trigger = "phy0tpt";
                };
```
(since github didn't track the change so well in the diff)

mpratt14 commented:
2 things, in my opinion 

we should come up with a unique name for the dtsi, like "engenius_ews-dual-ap"

everything in the root node in each dts should be left in the dts (aliases, keys, leds) instead of moved to a dtsi

mpratt14 commented:
i did a similar thing for eap600 if you would like to refer to those dts/dtsi

abraha2d commented:
> we should come up with a unique name for the dtsi, like "engenius_ews-dual-ap"
probably good, didn't really know what to call it. Settled on just `ews660ap` because they all share that same FCC ID.

> everything in the root node in each dts should be left in the dts (aliases, keys, leds) instead of moved to a dtsi
Am fairly new to device tree stuff... what's the reasoning behind this?

DragonBluep commented on target/linux/ath79/dts/qca9558_engenius_ews660ap.dtsi:
```diff
@@ -0,0 +1,133 @@
+};
+
+&ath10k_1 {
+	status = "okay";
```
`status = "okay";` is useless as this node is default enabled.

DragonBluep commented on target/linux/ath79/image/generic.mk:
```diff
@@ -1540,6 +1540,13 @@ define Device/engenius_ews660ap
 TARGET_DEVICES += engenius_ews660ap
 
+define Device/engenius_ens1750
+  $(Device/engenius_ews660ap)
```
It's better to create a common definition for ews660ap and ens1750 instead of calling `Device/engenius_ews660ap` directly. There are a lot of examples in this file.

abraha2d replied:
Created `Device/engenius_dual_ap` [here](https://github.com/openwrt/openwrt/blob/372f66a2b5c68b469c31081af2686ff036309aaf/target/linux/ath79/image/generic.mk#L1531). Is that what you meant?

DragonBluep replied:
Yes. LGTM.

mpratt14 commented:
> > we should come up with a unique name for the dtsi, like "engenius_ews-dual-ap"
> 
> Sounds good, will rename.
> 
> 
> > everything in the root node in each dts should be left in the dts (aliases, keys, leds) instead of moved to a dtsi
> 
> Am fairly new to device tree stuff... what's the reasoning behind this?

basically we want the dtsi to be as general as possible. like you said, you would like to add the ENS1200, and ideally, we can use the same dtsi for that one as well, and if that's possible then a name based on the FCC ID wouldn't be suitable.

also I have witnessed devices with what looks like the exact same board design that I intend to use the same dtsi for, and it turns out that they have different colored LEDs or one of the LEDs or buttons is on a different gpio line

abraha2d commented:
> basically we want the dtsi to be as general as possible. like you said, you would like to add the ENS1200, and ideally, we can use the same dtsi for that one as well, and if that's possible then a name based on the FCC ID wouldn't be suitable.

I see. Btw, the ENS1200 also has the same FCC ID (A8J-EWS660AP). Does it make sense to keep this dtsi as is?
My thought process was "this is the EWS660AP, and these are the other things that are literally exactly the same, but have different model numbers".

abraha2d commented:
Hm, flashed an image built off `b4a789a`, and it looks like 2.4G wifi is not recognized?
dmesg: https://pastebin.com/bm1TALcP
fwiw, 2.4G was working in the initial build based off v23.05.3 instead of main (don't have the commit hash, github ate it).

DragonBluep commented:
Please check if `ath10k` packages are missing. You can run `rm .config* && make prereq` to rebuild a clean image.

> Hm, flashed an image built off `b4a789a`, and it looks like the 5GHz radio is not recognized?
> 
> Shows up as "Generic unknown / Device is not active"
> 
> dmesg: https://pastebin.com/bm1TALcP
> 
> fwiw, 5GHz was working in the initial build based off v23.05.3 instead of main (don't have the commit hash, github ate it).

abraha2d commented:
Yup that was it. Working again with kmod-ath10k, ath10k-board-qca988x, and ath10k-firmware-qca988x.

abraha2d commented:
Anything else that I need to do for this?
